pkg/applicationserver/io/formatters/json_test.go

```diff
@@ -81,10 +81,12 @@ func TestJSONUpstream(t *testing.T) {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
 			a := assertions.New(t)
 			buf, err := formatter.FromUp(tc.Message)
-			if !a.So(err, should.BeNil) {
-				t.FailNow()
+			for _, v := range buf {
+				if !a.So(err, should.BeNil) {
+					t.FailNow()
+				}
+				a.So(string(v), should.Equal, tc.Result)
 			}
-			a.So(string(buf), should.Equal, tc.Result)
```

Comment:
Instead of ranging, assert that the length of the result is 1, and compare with `tc.Result` only the element at position `0`.

Reply:
Should I use, 
```
if len(buf) == 1{      
}
```
for the length assertion? or is there any alternate approach?

Reply:
`a.So(buf, should.HaveLength, 1) ` and `a.So(string(buf[0]), should.Equal, tc.Result)`